tests/python_package_test/test_engine.py

```diff
@@ -48,2591 +49,2655 @@ def categorize(continuous_x):
+
+
+def test_multiclass():
+    X, y = load_digits(n_class=10, return_X_y=True)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
+    params = {
+        'objective': 'multiclass',
+        'metric': 'multi_logloss',
+        'num_class': 10,
+        'verbose': -1
+    }
+    lgb_train = lgb.Dataset(X_train, y_train, params=params)
+    lgb_eval = lgb.Dataset(X_test, y_test, reference=lgb_train, params=params)
+    evals_result = {}
+    gbm = lgb.train(params, lgb_train,
+                    num_boost_round=50,
+                    valid_sets=lgb_eval,
+                    verbose_eval=False,
+                    evals_result=evals_result)
+    ret = multi_logloss(y_test, gbm.predict(X_test))
+    assert ret < 6
+    assert evals_result['valid_0']['multi_logloss'][-1] == pytest.approx(ret)
+
+
+def test_multiclass_rf():
+    X, y = load_digits(n_class=10, return_X_y=True)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
+    params = {
+        'boosting_type': 'rf',
+        'objective': 'multiclass',
+        'metric': 'multi_logloss',
+        'bagging_freq': 1,
+        'bagging_fraction': 0.6,
+        'feature_fraction': 0.6,
+        'num_class': 10,
+        'num_leaves': 50,
+        'min_data': 1,
+        'verbose': -1,
+        'gpu_use_dp': True
+    }
+    lgb_train = lgb.Dataset(X_train, y_train, params=params)
+    lgb_eval = lgb.Dataset(X_test, y_test, reference=lgb_train, params=params)
+    evals_result = {}
+    gbm = lgb.train(params, lgb_train,
+                    num_boost_round=50,
+                    valid_sets=lgb_eval,
+                    verbose_eval=False,
+                    evals_result=evals_result)
+    ret = multi_logloss(y_test, gbm.predict(X_test))
+    assert ret < 3
+    assert evals_result['valid_0']['multi_logloss'][-1] == pytest.approx(ret)
+
+
+def test_multiclass_prediction_early_stopping():
+    X, y = load_digits(n_class=10, return_X_y=True)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
+    params = {
+        'objective': 'multiclass',
+        'metric': 'multi_logloss',
+        'num_class': 10,
+        'verbose': -1
+    }
+    lgb_train = lgb.Dataset(X_train, y_train, params=params)
+    gbm = lgb.train(params, lgb_train,
+                    num_boost_round=50)
+
+    pred_parameter = {"pred_early_stop": True,
+                      "pred_early_stop_freq": 5,
+                      "pred_early_stop_margin": 1.5}
+    ret = multi_logloss(y_test, gbm.predict(X_test, **pred_parameter))
+    assert ret < 8
+    assert ret > 0.6  # loss will be higher than when evaluating the full model
+
+    pred_parameter = {"pred_early_stop": True,
+                      "pred_early_stop_freq": 5,
+                      "pred_early_stop_margin": 5.5}
+    ret = multi_logloss(y_test, gbm.predict(X_test, **pred_parameter))
+    assert ret < 2
```

Comment:
```suggestion
    assert ret < 0.2
```

from https://www.textcompare.org/index.html?id=6008e19c6d59840017ac7a56

![image](https://user-images.githubusercontent.com/7608904/105270731-e65df180-5b5b-11eb-9b12-c36dab66e39b.png)